mne/epochs.py

```diff
@@ -1546,6 +1549,34 @@ def equalize_event_counts(self, event_ids, method='mintime', copy=True):
             epochs.drop_bad_epochs()
         # figure out how to equalize
         eq_inds = list()
+
+        # deal with hierarchical tags
+        ids = epochs.event_id
+        if "/" in "".join(ids):
+            event_ids = [[x] if isinstance(x, string_types) else x
+                         for x in event_ids]
+            event_ids = [[k for k in ids if all((tag in k.split("/")
+                         for tag in id_))]
+                         if all(id__ not in ids for id__ in id_)
+                         else id_
+                         for id_ in event_ids]
+            for id in event_ids:
+                if len(set([id_ in ids for id_ in id])) != 1:
+                    err = ("Don't mix hierarchical and regular event_ids"
+                           " like in \'%s\'." % ", ".join(id))
+                    raise KeyError(err)
+
+            # deal with non-orthogonal tags
+            events_ = [set(epochs[x].events[:, 0]) for x in event_ids]
+            doubles = events_[0].intersection(events_[1])
+            if len(doubles):
+                warnings.warn("Warning: the two sets of epochs are "
+                              "overlapping. The %s overlapping epochs will"
+                              " be dropped." % len(doubles))
+                drop_ids = [ii for ii, t in enumerate(epochs.events[:, 0])
+                            if t in doubles]
+                epochs.drop_epochs(drop_ids, reason='EQUALIZED_COUNT')
```

Comment:
It's weird that two rounds of dropping are now done. Can't it be unified so that only one round of dropping is ever done? i.e., form your sets of epochs / indices to equalize (maybe in two or separate steps if necessary), then equalize in one place. We want this as DRY as possible.